src/elm/arithmetic.coffee

```diff
@@ -243,18 +274,38 @@ module.exports.Successor = class Successor extends Expression
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
+    successor = null
     if (not arg?)
       null
     else
-      MathUtil.successor arg
+      try
+        # MathUtil.successor throws on underflow, and the exception is used in
+        # the logic for evaluating `meets`
+        successor = MathUtil.successor arg
+      catch e
+        if e instanceof MathUtil.OverFlowException
+          null
```

Comment:
Again, based on the comment above, I don't know if we should actually be re-throwing this, but... as the code is now, I don't see the purpose of the `null` line above.  It's not the last line in the function, so it will not return.  And it is not assigned to anything.  So... it seems like a no-op?

Reply:
Good catch--it should be `return null` here.